packages/core/src/baseclient.ts

```diff
@@ -225,19 +213,22 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
   /**
    * @inheritDoc
    */
-  public getTransport(): Transport {
+  public getTransport(): NewTransport | undefined {
     return this._transport;
   }
 
   /**
    * @inheritDoc
    */
   public flush(timeout?: number): PromiseLike<boolean> {
-    return this._isClientDoneProcessing(timeout).then(clientFinished => {
-      return this.getTransport()
-        .close(timeout)
-        .then(transportFlushed => clientFinished && transportFlushed);
-    });
+    const transport = this._transport;
```

Comment:
I don't know for sure but one reason might be to prevent a race condition. We call flush on `transport` in a callback two lines below. If something were to set `this._transport` to `undefined` in between the `if (transport) {` and the callback execution it would result in an error.

Other thoughts @AbhiPrasad ?

Reply:
The race condition shouldn't exist, as there is no promise that executes in between, and we don't exit a closure, so there should be no event loop tick.

I actually did this for bundle size, but considering private members get minified, probably doesn't save that much. I'm just going to leave this now though, we can come back to it later.